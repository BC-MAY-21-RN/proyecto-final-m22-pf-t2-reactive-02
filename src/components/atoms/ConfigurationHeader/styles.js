import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    height: 250,
    backgroundColor: '#DAC3DB',
    borderBottomLeftRadius: 30,
    borderBottomRightRadius: 30,
    padding: 20,
  },
  image: {
    borderRadius: 90,
    width: 150,
    height: 150,
    alignSelf: 'center',
  },
  name: {
    fontSize: 25,
    alignSelf: 'center',
    fontWeight: 'bold',
  },
  header: {
    padding: 30,
  },
  changeImage: {
    backgroundColor: '#F1F1F1',
    position: 'absolute',
    right: 120,
    top: 120,
    borderRadius: 25,
    padding: 10,
  },
});

export default styles;
