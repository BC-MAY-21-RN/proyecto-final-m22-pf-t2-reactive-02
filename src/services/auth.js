import auth from '@react-native-firebase/auth';
import {GoogleSignin} from '@react-native-google-signin/google-signin';

function forgetPassword(text) {
  return auth().sendPasswordResetEmail(text);
}

function getId() {
  return auth().currentUser.uid;
}

function getName() {
  return auth().currentUser.displayName;
}

function getEmail() {
  return auth().currentUser.email;
}

function getPhoto() {
  return auth().currentUser.photoURL;
}

async function loginAccount(email, password) {
  return auth().signInWithEmailAndPassword(email, password);
}

function logout() {
  auth()
    .signOut()
    .then(() => {});
}

async function registerEmailUser(user) {
  return auth().createUserWithEmailAndPassword(
    user.object.email,
    user.object.password,
  );
}

async function onGoogleButtonPress() {
  const {idToken} = await GoogleSignin.signIn();
  const googleCredential = auth.GoogleAuthProvider.credential(idToken);
  return auth().signInWithCredential(googleCredential);
}

function updateUser(user, getPhotoUrl) {
  const update = {
    displayName: user.object.name,
    photoURL: getPhotoUrl(auth().currentUser.photoURL),
  };
  auth().currentUser.updateProfile(update);
}

export default {
  forgetPassword,
  getId,
  getName,
  getEmail,
  getPhoto,
  loginAccount,
  logout,
  onGoogleButtonPress,
  registerEmailUser,
  updateUser,
};
