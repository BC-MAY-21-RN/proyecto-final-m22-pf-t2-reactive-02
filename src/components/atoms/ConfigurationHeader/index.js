import React from 'react';
import {View, Text, Image, TouchableOpacity} from 'react-native';
import styles from '../ProfileHeader/styles';
import styleContainer from './styles';
import Icons from '../Icons';
import InputConfiguration from '../inputConfiguration';
import * as ImagePicker from 'react-native-image-picker';
import PickImage from '../../../assets/icons/add_a_photo.svg';

export default function ConfigurationHeader({image, name}) {
  const ChangeImage = () => {
    ImagePicker.launchImageLibrary();
  };
  return (
    <View style={styleContainer.container}>
      <Image
        style={styles.image}
        source={{
          uri: image,
        }}
      />
      <View style={{...styles.changeImage, ...{backgroundColor: '#000000'},}}>
        <TouchableOpacity onPress={ChangeImage}>
          <Icons IconProp={PickImage} />
        </TouchableOpacity>
      </View>
      <Text style={styles.name}>{name}</Text>
    </View>
  );
}
