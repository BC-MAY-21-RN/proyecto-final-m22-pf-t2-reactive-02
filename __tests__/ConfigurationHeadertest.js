import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ConfigurationHeader from '../src/components/atoms/ConfigurationHeader';
import '@react-native-firebase/auth';
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');
jest.mock('@react-native-google-signin/google-signin', () => {});

describe('Test from ConfigurationHeader component', () => {
  it('ConfigurationHeader snapshot test', () => {
    const renderConfigurationHeader = renderer
      .create(
        <ConfigurationHeader
          image={''}
          name={''}
          photo={''}
          setPhoto={() => {}}
        />,
      )
      .toJSON();
    expect(renderConfigurationHeader).toMatchSnapshot();
  });
});
