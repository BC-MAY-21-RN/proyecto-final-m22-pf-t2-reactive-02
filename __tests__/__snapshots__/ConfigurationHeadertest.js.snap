// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test from ConfigurationHeader component ConfigurationHeader snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#DAC3DB",
      "borderBottomLeftRadius": 30,
      "borderBottomRightRadius": 30,
      "height": 250,
      "padding": 20,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "borderRadius": 90,
        "height": 150,
        "width": 150,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#979797",
        "borderRadius": 25,
        "padding": 10,
        "position": "absolute",
        "right": 120,
        "top": 120,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View>
        < />
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 25,
        "fontWeight": "bold",
      }
    }
  />
</View>
`;
